core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -390,7 +394,16 @@ protected void validateDataFilesExist(TableMetadata base, Long startingSnapshotI
   @Override
   public Object updateEvent() {
     long snapshotId = snapshotId();
-    long sequenceNumber = ops.refresh().snapshot(snapshotId).sequenceNumber();
+    Snapshot justSaved = ops.refresh().snapshot(snapshotId);
+    long sequenceNumber = TableMetadata.INVALID_SEQUENCE_NUMBER;
```

Comment:
Well, I am not sure what should be the best way to react here. We do expect the catalog to be consistent so one option is to just log an error and not throw an event or throw some special even type. 

cc @rdsr who implemented this part and @rdblue who uses the event-based system. What do you expect in this case?

Reply:
What is the situation where the snapshot would not be in metadata? If the snapshot wasn't committed, then I would expect to not send an event.

Reply:
Nevermind, I see the comment about not being able to load the metadata.

I'd probably prefer null, but -1 is fine so that we don't change a public API.